inventory.go

```diff
@@ -0,0 +1,21 @@
+// Copyright 2016 Mender Software AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+package main
+
+type InventoryAttribute struct {
+	Name  string      `json:"name"`
+	Value interface{} `json:"value"`
+}
+
+type InventoryData []InventoryAttribute
```

Comment:
I think having here a `type InventoryData map[string][]InventoryAttribute` or even `type InventoryData map[string][]string` would simplify implementation a lot. Then whole reading part could be done as: 

```
type InventoryData map[string][]interface{}

func (id InventoryData) Write(p []byte) (n int, err error) {
	r := bufio.NewReader(bytes.NewBuffer(p))
	for {
		data, err := r.ReadBytes('\n')
		if err == io.EOF {
			return len(p), nil
		} else if err != nil {
			return 0, err
		}
		ia, _ := readAttr(data[:len(data)-1])
		id[ia.Name] = append(id[ia.Name], ia.Value)
	}
}

func readAttr(p []byte) (*InventoryAttribute, error) {
	val := strings.SplitN(string(p), "=", 2)
	// TOOD: check for errors
	return &InventoryAttribute{val[0], val[1]}, nil
}
```

Having this you can read data using something like:
```
mr := io.MultiReader(listRunnable) //os.File implements io.Reader
io.Copy(new(InventoryData), mr) //now InventoryData implements io.Writer
```